### ISSUE TITLE
[tabs] Various issues

### ISSUE THREAD

**Issue by DreierF** [sapphi-red/vitepress-plugins#23]
Hi,
First of all thanks for providing a tabs plugin for vitepress!

I tried it out in an effort to migrate an existing documentation from VuePress 1 to Vitepress/trying to find a replacement for [vuepress-plugin-tabs-less](https://tortorse.github.io/vuepress-plugin-tabs-less/). I ran into a couple of issues. Let me know if you would prefer separate issues for them, but to me it looks like the plugin is currently pretty broken, so not sure which of these make sense to be fixed independently.

Things I noticed:
- Using spaces in the title of the tabs is currently not supported, which is IMHO something that should be supported
- Using more advanced Markdown features such as a `:::tip` section within a tab also causes parser errors
- Using only the [example from the documentation](https://vitepress-plugins.sapphi.red/tabs/#tabs-with-non-shared-selection-state) leads to the following error at runtime (causing a blank page)
```
useTabs.ts:39 Uncaught (in promise) Error: [vitepress-plugin-tabs] TabsSharedState should be injected
    at useTabs (useTabs.ts:39:11)
    at setup (PluginTabs.vue:12:30)
    at callWithErrorHandling (chunk-WICYTZ2N.js?v=af802301:1565:18)
    at setupStatefulComponent (chunk-WICYTZ2N.js?v=af802301:8631:25)
    at setupComponent (chunk-WICYTZ2N.js?v=af802301:8592:36)
    at mountComponent (chunk-WICYTZ2N.js?v=af802301:6997:7)
    at processComponent (chunk-WICYTZ2N.js?v=af802301:6963:9)
    at patch (chunk-WICYTZ2N.js?v=af802301:6436:11)
    at mountChildren (chunk-WICYTZ2N.js?v=af802301:6680:7)
    at mountElement (chunk-WICYTZ2N.js?v=af802301:6587:7)
```

Reproduction: https://stackblitz.com/edit/github-zbjrsg-ytw6ky?file=index.md

**Comment by sapphi-red** [comment]
Hi

> Using spaces in the title of the tabs is currently not supported, which is IMHO something that should be supported

I simply didn't implement that as I created these plugins as an experiment. Contributions are welcome.

> Using more advanced Markdown features such as a :::tip section within a tab also causes parser errors

Tabs are already a container thus I didn't allow them to exist inside the tab.

> Using only the [example from the documentation](https://vitepress-plugins.sapphi.red/tabs/#tabs-with-non-shared-selection-state) leads to the following error at runtime (causing a blank page)

It seems `vite.optimizeDeps.exclude` is required.
```ts
export default defineConfig({
  vite: {
    optimizeDeps: {
      exclude: ['vitepress-plugin-tabs']
    }
  }
})
```
https://stackblitz.com/edit/github-zbjrsg-cg6vvk?file=.vitepress%2Fconfig.ts,.vitepress%2Ftheme%2Findex.ts

I guess it would fixed if this Symbol is changed into a string.
https://github.com/sapphi-red/vitepress-plugins/blob/7f4c8fe683650f6e8b82addc92c187fe07801fd4/packages/vitepress-plugin-tabs/src/client/useTabs.ts#L6

**Comment by DreierF** [comment]
Fixed with #24
